protocol/x/affiliates/keeper/keeper_test.go

```diff
@@ -394,6 +394,229 @@ func TestRegisterAffiliateEmitEvent(t *testing.T) {
+				Tiers: []types.AffiliateWhitelist_Tier{
+					{
+						Addresses:        []string{constants.AliceAccAddress.String()},
+						TakerFeeSharePpm: 250_000, // 25%
+					},
+					{
+						Addresses:        []string{constants.AliceAccAddress.String()},
+						TakerFeeSharePpm: 350_000, // 35%
+					},
+				},
+			},
+			expectedError: types.ErrDuplicateAffiliateAddressForWhitelist,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			err := k.SetAffiliateWhitelist(ctx, tc.whitelist)
+			if tc.expectedError != nil {
+				require.ErrorIs(t, err, tc.expectedError)
+			} else {
+				require.NoError(t, err)
+
+				storedWhitelist, err := k.GetAffiliateWhitelist(ctx)
+				require.NoError(t, err)
+				require.Equal(t, tc.whitelist, storedWhitelist)
+			}
+		})
+	}
+}
+
+func TestGetAffiliateWhiteListMap(t *testing.T) {
+	testCases := []struct {
+		name           string
+		whitelist      *types.AffiliateWhitelist
+		expectedLength int
+		expectedMap    map[string]uint32
+	}{
+		{
+			name: "Multiple tiers with multiple addresses",
+			whitelist: &types.AffiliateWhitelist{
+				Tiers: []types.AffiliateWhitelist_Tier{
+					{
+						Addresses:        []string{constants.AliceAccAddress.String(), constants.CarlAccAddress.String()},
+						TakerFeeSharePpm: 100_000, // 10%
+					},
+					{
+						Addresses:        []string{constants.BobAccAddress.String()},
+						TakerFeeSharePpm: 200_000, // 20%
+					},
+				},
+			},
+			expectedLength: 3,
+			expectedMap: map[string]uint32{
+				constants.AliceAccAddress.String(): 100_000, // 10%
+				constants.CarlAccAddress.String():  100_000, // 10%
+				constants.BobAccAddress.String():   200_000, // 20%
+			},
+		},
+		{
+			name: "Single tier with single address",
+			whitelist: &types.AffiliateWhitelist{
+				Tiers: []types.AffiliateWhitelist_Tier{
+					{
+						Addresses:        []string{constants.AliceAccAddress.String()},
+						TakerFeeSharePpm: 150_000, // 15%
+					},
+				},
+			},
+			expectedLength: 1,
+			expectedMap: map[string]uint32{
+				constants.AliceAccAddress.String(): 150_000, // 15%
+			},
+		},
+		{
+			name: "Empty tiers",
+			whitelist: &types.AffiliateWhitelist{
+				Tiers: []types.AffiliateWhitelist_Tier{},
+			},
+			expectedLength: 0,
+			expectedMap:    map[string]uint32{},
+		},
+		{
+			name:           "tiers not set",
+			whitelist:      nil,
+			expectedLength: 0,
+			expectedMap:    map[string]uint32{},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			ctx, k, _, _ := keepertest.AffiliatesKeepers(t, true)
+
+			if tc.whitelist != nil {
+				err := k.SetAffiliateWhitelist(ctx, *tc.whitelist)
+				require.NoError(t, err)
+			}
+
+			whitelistMap, err := k.GetAffiliateWhitelistMap(ctx)
+			require.NoError(t, err)
+			require.Equal(t, tc.expectedLength, len(whitelistMap))
+			require.Equal(t, tc.expectedMap, whitelistMap)
+		})
+	}
+}
+
+func TestGetTakerFeeShareViaWhitelist(t *testing.T) {
```

Comment:
Do we have a unit test where the affiliate qualifies through regular affiliate program and gets overridden to the higher whitelist fee share?

Reply:
Yes, the first unit test here registers an affiliate(which should have revshare of 15%) and we bump it to 40%